rholang/src/test/scala/coop/rchain/rholang/interpreter/storage/ChargingRSpaceTest.scala

```diff
@@ -0,0 +1,248 @@
+package coop.rchain.rholang.interpreter.storage
+
+import cats.Id
+import cats.effect.Sync
+import com.google.protobuf.ByteString
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.models.Channel.ChannelInstance.{ChanVar, Quote}
+import coop.rchain.models.TaggedContinuation.TaggedCont.ParBody
+import coop.rchain.models.Var.VarInstance.FreeVar
+import coop.rchain.models._
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.rholang.interpreter.Runtime.{RhoContext, RhoISpace, RhoPureSpace}
+import coop.rchain.rholang.interpreter.accounting.{CostAccount, CostAccountingAlg, _}
+import coop.rchain.rholang.interpreter.errors
+import coop.rchain.rholang.interpreter.errors.OutOfPhlogistonsError
+import coop.rchain.rholang.interpreter.storage.ChargingRSpaceTest._
+import coop.rchain.rspace
+import coop.rchain.rspace._
+import coop.rchain.rspace.history.Branch
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalactic.TripleEqualsSupport
+import org.scalatest.{Matchers, Outcome, fixture}
+
+import scala.collection.immutable
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class ChargingRSpaceTest extends fixture.FlatSpec with TripleEqualsSupport with Matchers {
+
+  behavior of "ChargingRSpace"
+
+  it should "charge for storing data in tuplespace" in { fixture =>
+    val TestFixture(chargingRSpace, costAlg, pureRSpace) = fixture
+    // format: off
+    val channels = channelsN(1)
+    val patterns = patternsN(1)
+    val cont = continuation()
+    // format: on
+    val storageCost  = ChargingRSpace.storageCostConsume(channels, patterns, cont)
+    val minimumPhlos = storageCost + Cost(1)
```

Comment:
Hmm.. If there was a match it will stick because charging for a match is the last thing in the process. There is a test in my last PR that might be what you're looking for. The `+ Cost(1)` is needed here to have a _bufor_.

Reply:
> If there was a match it will stick because charging for a match is the last thing in the process.

I don't quite get it.

Why is that buffer needed? Is it?

If the `+1` is needed (then I'd like to understand why, preferably from reading a variable name), or its not and should be removed from the test (so that it doesn't suggest false requirements)